data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/ConsumerOffsetManager.java

```diff
@@ -0,0 +1,42 @@
+package dev.knative.eventing.kafka.broker.dispatcher;
+
+import io.vertx.kafka.client.consumer.KafkaConsumerRecord;
+
+public interface ConsumerOffsetManager<K, V> {
```

Comment:
Yes, it's a Strategy pattern.

> Because you could use such strategy for records with different generics, it seems to me that you should push the generics down to the single methods and not at interface level

when would this happen?

Reply:
> when would this happen?

It's not about when happen, is about reasoning about the types in order to understand the code :smile: You use a generic at interface level to identify a type property which is "discriminatory" of the implementation itself, while in that case it's not because these strategy implementation are potentially valid for every K, V out there

Reply:
I'll do it.

Reply:
Sorry, I'm trying to do it, but I need to know `<K, V>` at the object level to inject `KafkaConsumer<K, V>` to the implementation `UnorderedConsumerOffsetManager`.